examples/jetstream/consumers/main.go

```diff
@@ -0,0 +1,181 @@
+`-push-target` option can be used to configure a delivery subject for a push-based consumer. If not provided, the consumer will be created in push mode.
+`-filter` option can be used to filter subjects from the stream.
+`-queue` option can be used to create a queue consumer.
+
+Example usage:
+./js-consumer-management -creds '/path/to/nats/credentials' -s 'nats://127.0.0.1:4222' -push-target=ORDERS_RECV add ORDERS ORDERS_CONSUMER
+*/
+
+func usage() {
+	log.Printf("Usage: js-consumer-management [-s server] [-creds file] [-nkey file] [-tlscert file] [-tlskey file] [-tlscacert file] [-push-target subject] [-filter subject] [-queue queue] <operation> [ add | delete | info ] <stream> <name>\n")
+	flag.PrintDefaults()
+}
+
+func showUsageAndExit(exitcode int) {
+	usage()
+	os.Exit(exitcode)
+}
+
+func main() {
+
+	// CLI flags
+	var (
+		urls          = flag.String("s", nats.DefaultURL, "The nats server URLs (separated by comma)")
+		userCreds     = flag.String("creds", "", "User Credentials File")
+		nkeyFile      = flag.String("nkey", "", "NKey Seed File")
+		tlsClientCert = flag.String("tlscert", "", "TLS client certificate file")
+		tlsClientKey  = flag.String("tlskey", "", "Private key file for client certificate")
+		tlsCACert     = flag.String("tlscacert", "", "CA certificate to verify peer against")
+		pushTarget    = flag.String("push-target", "", "Deliverty subject for push-based consumer")
+		filter        = flag.String("filter", "", "FilterSubject for a consumer")
+		queue         = flag.String("queue", "", "Delivery target group for a consumer")
+		showHelp      = flag.Bool("h", false, "Show help message")
+	)
+
+	log.SetFlags(0)
+	flag.Usage = usage
+	flag.Parse()
+
+	if *showHelp {
+		showUsageAndExit(0)
+	}
+
+	args := flag.Args()
+	if len(args) != 3 {
+		showUsageAndExit(1)
+	}
+
+	// Connect Options.
+	opts := []nats.Option{nats.Name("JetStream Sample Stream Management")}
+
+	if *userCreds != "" && *nkeyFile != "" {
+		log.Fatal("specify -seed or -creds")
+	}
+
+	// Use UserCredentials
+	if *userCreds != "" {
+		opts = append(opts, nats.UserCredentials(*userCreds))
+	}
+
+	// Use TLS client authentication
+	if *tlsClientCert != "" && *tlsClientKey != "" {
+		opts = append(opts, nats.ClientCert(*tlsClientCert, *tlsClientKey))
+	}
+
+	// Use specific CA certificate
+	if *tlsCACert != "" {
+		opts = append(opts, nats.RootCAs(*tlsCACert))
+	}
+
+	// Use Nkey authentication.
+	if *nkeyFile != "" {
+		opt, err := nats.NkeyOptionFromSeed(*nkeyFile)
+		if err != nil {
+			log.Fatal(err)
+		}
+		opts = append(opts, opt)
+	}
+
+	// Connect to NATS
+	nc, err := nats.Connect(*urls, opts...)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer nc.Close()
+
+	// Retrieve JetStream connections
+	js, err := nc.JetStream()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	operation, streamName, consumerName := args[0], args[1], args[2]
+
+	switch operation {
+	case "add":
+		// create new stream if it does not exist
```

Comment:
Recommend a hard error if the stream does not exist.   Otherwise, the example could confuse developer around the difference between stream management and consumer management.